src/components/wyehome/OutOfOfficeEdit.jsx

```diff
@@ -0,0 +1,254 @@
+        finish_at: `${toDate}T23:59:59`,
+        user_id: userId,
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          const { onModeChange, showAlert } = this.props;
+          showAlert('Out of Office was updated!', 'info');
+          onModeChange('list');
+          return null;
+        }
+        return response.json();
+      }).then((json) => {
+        if (json !== null) {
+          const { showAlert } = this.props;
+          showAlert(json[0], 'warning');
+        }
+      }).catch(() => {
+        const { onModeChange, showAlert } = this.props;
+        showAlert('An error occured while updating Out of Office. Try again later.', 'warning');
+        onModeChange('list');
+      });
+  };
+
+  toggle() {
+    const { showModal } = this.state;
+    this.setState({ showModal: !showModal });
+  }
+
+  modal() {
+    const { showModal } = this.state;
+    return (
+      <div>
+        <Modal isOpen={showModal} toggle={() => this.toggle()}>
+          <ModalHeader toggle={() => this.toggle()}>Confirm deletion</ModalHeader>
+          <ModalBody>
+             Do you want to delete this Out of Office?
+          </ModalBody>
+          <ModalFooter>
+            <Button color="primary" onClick={() => this.handleDelete()}>Confirm</Button>
+            {' '}
+            <Button color="secondary" onClick={() => this.toggle()}>Cancel</Button>
+          </ModalFooter>
+        </Modal>
+      </div>
+    );
+  }
+
+  render() {
+    const {
+      reason, minDate, fromDate, toDate,
+    } = this.state;
+    const { onModeChange } = this.props;
+    return (
+      <Form onSubmit={this.handleSubmit}>
+        { this.modal() }
+        <FormGroup>
+          <Label for="exampleSelect">Reason</Label>
+          <Select
+            id="reason-selection"
+            classNamePrefix="react_select"
+            placeholder="Reason..."
+            options={[
+              { label: 'Work from Home', value: 'atHome' },
+              { label: 'Travel', value: 'travel' },
+              { label: 'Vacations', value: 'vacations' },
+              { label: 'Sick', value: 'sick' },
+              { label: 'Others', value: 'other' },
+            ]}
+            value={reason}
+            onChange={selection => this.setState({ reason: selection })}
+          />
+        </FormGroup>
+        <FormGroup>
+          <Label>From</Label>
+          <Input type="date" name="fromDate" onChange={this.handleInput} min={minDate} defaultValue={fromDate} required />
+        </FormGroup>
+        <FormGroup>
+          <Label>To</Label>
+          <Input type="date" id="toDate" name="toDate" onChange={this.handleInput} min={fromDate} defaultValue={toDate} required />
+        </FormGroup>
+        <div className="d-flex form-buttons">
+          <Button className="mr-1" color="primary">Save</Button>
+          <Button className="ml-1" color="dark" onClick={() => this.toggle()}>Delete</Button>
```

Comment:
I would add a `mr-1` in the className so the spacing between the middle and right buttons is the same as between the middle and left buttons:
```suggestion
          <Button className="ml-1 mr-1" color="dark" onClick={() => this.toggle()}>Delete</Button>
```